paymail_service_provider.go

```diff
@@ -172,6 +173,22 @@ func (p *PaymailDefaultServiceProvider) RecordTransaction(ctx context.Context,
 		return nil, err
 	}
 
+	if p2pTx.DecodedBeef == nil {
+		return &paymail.P2PTransactionPayload{
+			Note: p2pTx.MetaData.Note,
+			TxID: transaction.ID,
+		}, nil
+	}
+
+	if reflect.TypeOf(rts) == reflect.TypeOf(&externalIncomingTx{}) {
+		for _, input := range p2pTx.DecodedBeef.InputsTxData {
+			err = saveBeefTransactionInput(ctx, p.client, input.Transaction)
```

Comment:
I would try to simplify. 
Get all transaction inputs, and try to proceed like this:

Pseudo code:
```go
var txIDs []string
 for _, input := range p2pTx.DecodedBeef.InputsTxData {
   txIDs = append(txIDs, input.UtxoPointer.TransactionID)
}
`txs, err := client.GetTransactionsByIDs(ctx, txIds)`
...
// check if we found all of the input txs in db
if len(txs) == len(txIds) {
   // we have all of them in db
   return 
}

// get the missing ones (compare txs and txIds)
// for each not found tx -> proceed your business logic

Reply:
Done